wasm/src/lib.rs

```diff
@@ -171,39 +171,63 @@ pub fn run(
 }
 
 fn run_kata_exercise_internal(
-    verification_source: &str,
-    exercise_implementation: &str,
+    exercise_code: &str,
+    solution_code: &str,
+    verification_code: &str,
+    code_dependencies: Vec<(SourceName, SourceContents)>,
     event_cb: impl Fn(&str),
-) -> Result<bool, Vec<stateless::Error>> {
-    verify_exercise(
-        vec![
-            ("exercise".into(), exercise_implementation.into()),
-            ("verifier".into(), verification_source.into()),
-        ],
-        &mut CallbackReceiver { event_cb },
-    )
+) -> bool {
+    let mut sources = vec![
+        ("exercise".into(), exercise_code.into()),
+        ("solution".into(), solution_code.into()),
+        ("verification".into(), verification_code.into()),
+    ];
+    for code_dependency in code_dependencies {
+        sources.push(code_dependency);
+    }
+    let mut out = CallbackReceiver { event_cb };
+    let result = verify_exercise(sources, &mut out);
```

Comment:
After reading through all this I wonder if a better name for `verify_exercise` would be `check_solution`. There are so many things called "verify exercise" / "run exercise" / "validate exercise" that I think it's worth a bit of cleanup to make them more specific.

Again feel free to ignore if you think the name doesn't quite fit.

Reply:
I like the name `check_solution`! I'll rename `verify_exercise`.

Reply:
I think I fixed all the run/test/validate/verify weirdness.